SpellCheck: Update Scintilla styles for new filetypes in Geany 2.1

eht16 opened the pull request:
The usual update of Scintilla styles for which spell checking should be performed.

I don't know the new filetypes well enough but guessed according to previous mappings :smile: .

b4n reviewed (changes requested):
What about COBOL?  It seems to have new specific styles now.

I'd also consider `SCE_HJ_TEMPLATELITERAL` and `SCE_HJA_TEMPLATELITERAL`, which IIUC are f-string-like stuff.

And you need to add the brand new `SCE_DART_STRINGEOL`, `SCE_ZIG_STRINGEOL`, `SCE_NIX_STRINGEOL` and `SCE_TOML_STRINGEOL` :slightly_smiling_face:

techee commented:
> @techee OK. I still believe it'd be more future-proof to have them listed

OK, in that case, should I update Geany style mappings with these?

b4n commented:
@techee Don't you think it makes it easier whenever the COBOL lexer against incompatible styles?  Which IIUC is the reason Scintilla started having its own styles.

If you do, yes, that'd be neat.  But it can wait after 2.1 as well, whichever.

eht16 commented:
> What about COBOL? It seems to have new specific styles now.

I added the specific COBOL styles, seems to work as expected.

> I'd also consider `SCE_HJ_TEMPLATELITERAL` and `SCE_HJA_TEMPLATELITERAL`, which IIUC are f-string-like stuff.

Ah thanks, added.

> And you need to add the brand new `SCE_DART_STRINGEOL`, `SCE_ZIG_STRINGEOL`, `SCE_NIX_STRINGEOL` and `SCE_TOML_STRINGEOL` 🙂

Done.
I'm not too sure if unterminated strings are worth for spell checking but since for other languages, they were already present, it's fine to add these too.

techee commented:
> @techee Don't you think it makes it easier whenever the COBOL lexer against incompatible styles? Which IIUC is the reason Scintilla started having its own styles.

Agree, see https://github.com/geany/geany/pull/4337.
